data_prep.py

```diff
@@ -1,150 +1,330 @@
+        )
+        self.df = self.merge_files()
+        self.concat_df_with_oh_encoding()
+
+    def read_csv_file(self, file_prefix) -> pd.DataFrame:
+        return pd.read_csv(
+            self.data_path / f"{file_prefix}_{self.data_type}.csv", sep=";"
+        )
+
+    def read_all_files(self) -> (pd.DataFrame, pd.DataFrame, pd.DataFrame):
+        bkg = (
+            self.read_csv_file("BKG").drop(columns=self.get_bkg_drop_cols())
+            if self.data_type == "train"
+            else self.read_csv_file("BKG")
+        )
+        logger.info(f"{datetime.now()} loaded booking data")
+        tkt = self.read_csv_file("TKT")
+        logger.info(f"{datetime.now()} loaded ticket data")
+        fcp = self.read_csv_file("FCP")
+        logger.info(f"{datetime.now()} loaded flight coupon data")
+        return (
+            bkg,
+            tkt,
+            fcp[fcp.UPGRADED_FLAG == "Y"]
+            if self.model_type == "predict_when_upgrade"
+            else fcp,
+        )
+
+    def merge_files(self) -> pd.DataFrame:
+        bkg, tkt, fcp = self.read_all_files()
+        logger.info(f"{datetime.now()} merging datasets")
+        return fcp.merge(tkt, on="TICKET_NUMBER", how="left").merge(
+            bkg, on="BOOKING_ID", how="left"
+        )
+
+    @staticmethod
+    def discard_nonupgraded_rows(fcp: pd.DataFrame) -> pd.DataFrame:
+        return fcp[fcp["UPGRADED_FLAG"] == "Y"]
+
+    @staticmethod
+    def get_bkg_drop_cols() -> list:
+        return ["UPGRADED_FLAG", "UPGRADE_TYPE", "UPGRADE_SALES_DATE"]
+
+    @staticmethod
+    def get_columns_to_drop() -> list:
+        return [
+            "TICKET_NUMBER",
+            "ORIGIN_AIRPORT_CODE",
+            "DESTINATION_AIRPORT_CODE",
+            "SALES_DATE",
+            "FLIGHT_DATE_LOCAL",
+            "MARKETING_CARRIER",
+            "OPERATIONAL_CARRIER",
+            "BOOKED_CLASS",
+            "AIRCRAFT_TYPE",
+            "FARE_BASIS",
+            "BOOKING_ID",
+            "ORIGINAL_TICKET_NUMBER",
+            "SEGMENTS",
+            "FLIGHT_COUPONS",
+            "FORM_OF_PAYMENT",
+            "CURRENCY",
+            "TOTAL_PRICE",
+            "LOYAL_CUSTOMER_ID",
+            "LOYAL_CUSTOMER_DATE_OF_BIRTH",
+            "LOYAL_CUSTOMER_REGISTERED_DATE",
+            "SALES_DATE",
+            "SALES_MARKET",
+            "SEGMENTS",
+            "INTINERARY",
+            "BOOKING_ORIGIN_AIRPORT",
+            "BOOKING_ORIGIN_COUNTRY_CODE",
+            "BOOKING_DEPARTURE_TIME_UTC",
+            "BOOKING_DESTINATION_AIRPORT",
+            "BOOKING_DESTINATION_COUNTRY_CODE",
+            "BOOKING_ARRIVAL_TIME_UTC",
+        ]
+
+    def get_cols_to_drop_for_training(self) -> list:
+        cols = ["UPGRADE_TYPE"]
+        cols.append(
+            "UPGRADE_SALES_DATE"
+        ) if self.model_type == "predict_upgrade" else cols.append("UPGRADED_FLAG")
+        return cols
+
+    def get_new_cols_for_predict_when_model(self) -> None:
```

Comment:
Maybe name like `prepare_new_cols` would be more clear?

Reply:
changed method name, made sure that it was used (accidentally ommited it in the code)